### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by yasiupl**
Weekly Issues Fix Dump

### CODE REVIEW

File: backend/server.js [hunk 1]
```diff
@@ -867,7 +639,7 @@ async function processData(data, query, callback) {
 
         /* else {
```
Comment (allgreed) [inline backend/server.js]: Can we delete the commented code altogether?

File: backend/src/scrape/agencyShortName.js [hunk 2]
```diff
@@ -0,0 +1,216 @@
+// Shortnames are used to take up less space on the frontend, while preserving maximum information about agency Name
+// Shortname is either a Full name when shorter than or equal to X characters and Abbreviation when longer than X characters.
+// X = 16
+
+// TODO: dynamicaly create shortNames array, given length treshold.
```
Comment (allgreed) [inline backend/src/scrape/agencyShortName.js]: Can we have an issue for this?

File: backend/src/scrape/agencyShortName.js [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+    "",
+    "ATK",
+    "BAC",
+    "Chrysler",
+    "Avio S.p.A",
+    "RAAF",
+    "USAF",
+    "PLA",
+    "VKO",
+    "",
```
Comment (allgreed) [inline backend/src/scrape/agencyShortName.js]: Why there are empty spaces here?

File: backend/tests/unit/scrape/agencyShortName.test.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+each([-100, 0, 100]).test('return "Unknown" on value outside reference table', (agencyID) => {
+   
+
+    const result = agencyType(agencyID, stub)
+
+    expect(result).toBe('Unknown');
+});
+
+each([1,2,3]).test('returns valid agency from agencyResolutionTable', (agencyID) => {
```
Comment (allgreed) [inline backend/tests/unit/scrape/agencyShortName.test.js]: Could you please put expected values inline with test examples?

File: backend/src/scrape/agencyType.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    "Government",
+    "Multinational",
+    "Commercial",
+    "Educational",
+    "Private",
+];
+
+function agencyType(agencyID, agenciesResolutionTable = agenciesType) 
+{
+    if(agencyID > 0 && agencyID - 1 < agenciesResolutionTable.length)
```
Comment (allgreed) [inline backend/src/scrape/agencyType.js]: For future reference (not suggesting a change, this is absolutely fine!): a ternary expression would shine here IMO ;)

File: backend/tests/unit/scrape/agencyShortName.test.js [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+each([undefined, null]).test('return "Unknown" on undefined/null', (agencyID) => {
+   
+
+    const result = agencyType(agencyID, stub)
+
+    expect(result).toBe('Unknown');
+});
+
+
+each([-100, 0, 100]).test('return "Unknown" on value outside reference table', (agencyID) => {
```
Comment (allgreed) [inline backend/tests/unit/scrape/agencyShortName.test.js]: How about adding 4 here, just to be safe?

File: backend/tests/unit/scrape/agencyType.test.js [hunk 7]
```diff
@@ -0,0 +1,34 @@
+const each = require('jest-each').default;
+
```
Comment (allgreed) [inline backend/tests/unit/scrape/agencyType.test.js]: If you're having the same tests for different submodules maybe an abstraction should emerge? Will those tests still pass after resolving the empty element issue?

File: backend/server.js (hunk 1 above)
Comment (yasiupl) [reply backend/server.js]: Yup.

File: backend/src/scrape/agencyShortName.js (hunk 3 above)
Comment (yasiupl) [reply backend/src/scrape/agencyShortName.js]: As mentioned before, deleted agencies from the database. The array preserves the ID mappings as scraped from the database. The function should return "Unknown" on those too.

File: backend/tests/unit/scrape/agencyShortName.test.js (hunk 6 above)
Comment (yasiupl) [reply backend/tests/unit/scrape/agencyShortName.test.js]: Sounds reasonable.

**Pull request merged**